extras-scalapb/src/main/scala/com/avast/clients/rabbitmq/extras/format/ScalaPBAsJsonProductConverter.scala

```diff
@@ -0,0 +1,42 @@
+package com.avast.clients.rabbitmq.extras.format
+
+import com.avast.bytes.Bytes
+import com.avast.clients.rabbitmq.ProductConverter
+import com.avast.clients.rabbitmq.api.{ConversionException, MessageProperties}
+import scalapb.GeneratedMessage
+import scalapb.json4s.Printer
+
+import scala.annotation.implicitNotFound
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+@implicitNotFound("Could not generate ScalaPBAsJsonProductConverter for ${A}, try to import or define some")
+trait ScalaPBAsJsonProductConverter[A <: GeneratedMessage] extends ProductConverter[A]
+
+object ScalaPBAsJsonProductConverter {
+  def derive[A <: GeneratedMessage: ScalaPBAsJsonProductConverter](): ScalaPBAsJsonProductConverter[A] =
+    implicitly[ScalaPBAsJsonProductConverter[A]]
+
+  // Printing is intentionally configured to use integer for enums
+  // because then the serialized data is valid even after values renaming.
+  private val jsonPrinter = new Printer().formattingEnumsAsNumber
```

Comment:
I'm sorry, I'm not familiar with `scalapb.json4s.Printer`, but it does print even the fields which have the default value, right?

Reply:
No, you have to specify `includingDefaultValueFields` to get that behavior.

Reply:
OK. I think we should do that. Without it, the users/consumers weren't happy with it and demanded those fields to be present.